### PR TITLE
getAllRelationshipData

### PR DESCRIPTION
**Pull request by RohiniKambam**
1.Created Types.
2.Created Relations between Types.
2.Using that Relationship getAll the relations.
[getAllrelationobjectsdata.txt](https://github.com/PogeyanOSS/cmis-swagger/files/2286474/getAllrelationobjectsdata.txt)

### CODE REVIEW

File: copper-swagger-api/src/main/java/com/pogeyan/swagger/api/utils/SwaggerHelpers.java [hunk 1]
```diff
@@ -1227,4 +1251,88 @@ public static String getHostSwaggerUrl() {
...
+
+	/**
+	 * 
+	 * @param httpRequest
+	 * @return username and password from the request
+	 * @throws UnsupportedEncodingException
+	 */
+
+	private static String getB64Auth(String username, String password) {
+		String source = username + ":" + password;
```
Comment (azarMohamed) [inline copper-swagger-api/src/main/java/com/pogeyan/swagger/api/utils/SwaggerHelpers.java]: Http Utilis reuse getBasic auth method

File: copper-swagger-api/src/main/java/com/pogeyan/swagger/api/utils/SwaggerHelpers.java [hunk 2]
```diff
@@ -1227,4 +1251,88 @@ public static String getHostSwaggerUrl() {
...
+	 *            repository Id
+	 * @param objectId
+	 *            objectId for which the relation data needs to be fetched
+	 * @return
+	 */
+	@SuppressWarnings("unchecked")
+	public static ArrayList<Object> getDescendantsForRelationObjects(String username, String password, String repoId,
+			String objectId) {
+		// TODO Auto-generated method stub
+		CloseableHttpClient httpclient = HttpClients.createDefault();
```
Comment (azarMohamed) [inline copper-swagger-api/src/main/java/com/pogeyan/swagger/api/utils/SwaggerHelpers.java]: Use proper variable name

File: copper-swagger-api/src/main/java/com/pogeyan/swagger/api/utils/SwaggerHelpers.java [hunk 3]
```diff
@@ -1227,4 +1251,88 @@ public static String getHostSwaggerUrl() {
...
+			getRequest.addHeader("Authorization", getB64Auth(username, password));
+			httpResponse = httpclient.execute(getRequest);
+			int resCode = httpResponse.getStatusLine().getStatusCode();
+			HttpEntity resEntity = httpResponse.getEntity();
+			if (resEntity != null) {
+				String resBody = EntityUtils.toString(resEntity);
+				ArrayList<Object> json = new ObjectMapper().readValue(resBody, ArrayList.class);
+				return json;
+			} else {
+				LOG.error("Empty ResponseEntity with resCode: {}", resCode);
```
Comment (azarMohamed) [inline copper-swagger-api/src/main/java/com/pogeyan/swagger/api/utils/SwaggerHelpers.java]: Use log format -> ClassName methodName repoId error

File: copper-swagger-api/src/main/java/com/pogeyan/swagger/api/utils/SwaggerHelpers.java [hunk 4]
```diff
@@ -1227,4 +1251,88 @@ public static String getHostSwaggerUrl() {
...
+		}
+		return null;
+	}
+
+	@SuppressWarnings("unchecked")
+	public static Map<String, Object> formRelationData(Session session, ArrayList<Object> relationData) {
+		Map<String, Object> relMap = new LinkedHashMap<String, Object>();
+		if (relationData != null) {
+			relationData.forEach(relationObj -> {
+				JSONObject jsonSub = new JSONObject();
```
Comment (azarMohamed) [inline copper-swagger-api/src/main/java/com/pogeyan/swagger/api/utils/SwaggerHelpers.java]: change variable name

File: copper-swagger-api/src/main/java/com/pogeyan/swagger/api/utils/SwaggerHelpers.java [hunk 5]
```diff
@@ -1227,4 +1251,88 @@ public static String getHostSwaggerUrl() {
...
+				LinkedHashMap<Object, Object> relationObjectMap = (LinkedHashMap<Object, Object>) relationObj;
+				LinkedHashMap<Object, Object> object1 = (LinkedHashMap<Object, Object>) relationObjectMap.get("object");
+				LinkedHashMap<Object, Object> object2 = (LinkedHashMap<Object, Object>) object1.get("object");
```
Comment (azarMohamed) [inline copper-swagger-api/src/main/java/com/pogeyan/swagger/api/utils/SwaggerHelpers.java]: relationObjectMap.get("object") !=null ? relationObjectMap.get("object") .get("object") :null;

File: copper-swagger-api/src/main/java/com/pogeyan/swagger/api/utils/SwaggerHelpers.java (hunk 1 above)
Comment (RohiniKambam) [reply copper-swagger-api/src/main/java/com/pogeyan/swagger/api/utils/SwaggerHelpers.java]: httpUtills method is not there in swagger,so we use this method.

**Pull request merged**